src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs

```diff
@@ -211,10 +211,18 @@ private async Task Initialize(CancellationToken cancellationToken)
             }
         });
 
-        if (_syncConfig.SnapSync)
+        if (_api.SyncModeSelector is MultiSyncModeSelector multiSyncModeSelector)
```

Comment:
I somewhat dislike this condition, would making it abstract (based on the interface for example) be hard/worse?